flutter/lib/screens/qr_code_screen.dart

```diff
@@ -0,0 +1,138 @@
+//Flutter packages
+import 'dart:async';
+import 'package:flutter/material.dart';
+//Third party packages
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:lottie/lottie.dart';
+import 'package:provider/provider.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import 'package:simple_connection_checker/simple_connection_checker.dart';
+//My Packages
+import '../providers/superformula_provider.dart';
+import 'offline_screen.dart';
+
+class QRCCodeScreen extends StatefulWidget {
+
+  static const routeName = "/qr_code_screen";
+  const QRCCodeScreen({Key? key}) : super(key: key);
+
+  @override
+  _QRCCodeScreenState createState() => _QRCCodeScreenState();
+}
+
+class _QRCCodeScreenState extends State<QRCCodeScreen> {
+
+  //Object declarations
+
+  late Timer _timer;
+  int _time = 0;
+  String _timeInString = "";
+
+  //Offline handler
+  StreamSubscription? _offlineStreamSubscription;
+
+  @override
+  void initState() {
+    super.initState();
+
+    //Simple connection checker object init
+    SimpleConnectionChecker _simpleConnectionChecker = SimpleConnectionChecker()
+      ..setLookUpAddress('pub.dev'); //Optional method to pass the lookup string
+    _offlineStreamSubscription = _simpleConnectionChecker.onConnectionChange.listen((connected) {
+      if(!connected){
+        //Launch offline screen when the device is offline
+        Navigator.of(context).pushNamedAndRemoveUntil(OfflineScreen.routeName, (route) => false);
+      }
+    });
+
+    //Call the seed API
+    var provider = Provider.of<SuperFormulaProvider>(context, listen: false);
+
+    provider.getSeed(firstTime: true,context: context);
+
+    //Timer to update time
+    _timer = Timer.periodic(const Duration(seconds: 1), (Timer t) async{
+
+      //Get NTP time
+      DateTime mySystemTime = DateTime.now().toUtc();
+
+      //Update the Seed[QR Code] every minute
+      if (mySystemTime.second==0) {
+        if(mounted) provider.getSeed(firstTime: false,context: context);
+      }
+
+      //Update the timer text
+      if (mounted) {
+        setState((){
+          //Update the time,Show time in decreasing order
+          _time = 59 - mySystemTime.second;
+
+          //Update the text
+          _timeInString = (_time).toString();
+        });
+      }
+    });
```

Comment:
**suggestion**: All this logic could be in its own provider. All you need is a stream to listen to or state broadcasted by the provider.